src/libultra/os/probetlb.s

```diff
@@ -17,70 +13,75 @@
  *  Returns the physical address if found, or -1 if not found.
  */
 LEAF(__osProbeTLB)
-    // Set C0_ENTRYHI based on supplied vaddr
-    mfc0    $t0, C0_ENTRYHI
-    andi    $t1, $t0, TLBHI_PIDMASK
-    li      $at, TLBHI_VPN2MASK
-    and     $t2, $a0, $at
-    or      $t1, $t1, $t2
-    mtc0    $t1, C0_ENTRYHI
-    nop
-    nop
-    nop
-    // TLB probe, sets C0_INX to a value matching C0_ENTRYHI.
-    // If no match is found the TLBINX_PROBE bit is set to indicate this.
-    tlbp
-    nop
-    nop
-    // Read result
-    mfc0    $t3, C0_INX
-    li      $at, TLBINX_PROBE
-    and     $t3, $t3, $at
-    // Branch if no match was found
-    bnez    $t3, 3f
-     nop
-    // Read TLB, sets C0_ENTRYHI, C0_ENTRYLO0, C0_ENTRYLO1 and C0_PAGEMASK for the TLB
-    // entry indicated by C0_INX
-    tlbr
-    nop
-    nop
-    nop
-    // Calculate page size = (page mask + 0x2000) >> 1
-    mfc0    $t3, C0_PAGEMASK
-    addi    $t3, $t3, 0x2000
-    srl     $t3, $t3, 1
-    // & with vaddr
-    and     $t4, $t3, $a0
-    // Select C0_ENTRYLO0 or C0_ENTRYLO1
-    bnez    $t4, 1f
-     addi   $t3, $t3, -1 // make bitmask out of page size
-    mfc0    $v0, C0_ENTRYLO0
+    /* Set C0_ENTRYHI based on supplied vaddr */
+    MFC0(   t0, C0_ENTRYHI)
+    and     t1, t0, TLBHI_PIDMASK
+.set noreorder; .set reorder
```

Comment:
Does this do something (set noreorder and than immediately set reorder)? Seems ultralib was able to match this file by just having the whole file be noreorder which seems better than noreorders scattered throughout?

Reply:
The noreorder toggle helped with instruction scheduling iirc. I'm generally very reluctant to just wrap a whole function in noreorder since usually that's not the right thing to do, but I think for the function it would actually be better overall, I'll change it